mapstruct-protobuf3/spi-impl/src/main/java/org/mapstruct/example/protobuf/ProtobufAccessorNamingStrategy.java

```diff
@@ -0,0 +1,33 @@
+package org.mapstruct.example.protobuf;
+
+import org.mapstruct.ap.spi.DefaultAccessorNamingStrategy;
+
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.DeclaredType;
+import javax.lang.model.type.TypeMirror;
+
+/**
+ * @author Thomas Kratz
+ */
+public class ProtobufAccessorNamingStrategy extends DefaultAccessorNamingStrategy {
+
+    @Override
+    public String getElementName(ExecutableElement adderMethod) {
+
+        String methodName = super.getPropertyName(adderMethod);
+        Element receiver = adderMethod.getEnclosingElement();
+        if (receiver != null) {
+            TypeElement type = (TypeElement) ((DeclaredType) receiver).asElement();
```

Comment:
I would rewrite the routine a little bit. Don't think its required go from a element to a `DeclaredType` to a `TypeElement` if you already know its a class. So like this:

```java
    @Override
    public String getElementName(ExecutableElement adderMethod) {

        String methodName = getPropertyName( adderMethod );
        Element receiver = adderMethod.getEnclosingElement();
        if ( receiver != null && receiver.getKind() == ElementKind.CLASS ) {
            TypeElement type = (TypeElement) receiver;
            TypeMirror superType = type.getSuperclass();
            if ( superType != null && PROTOBUF_MESSAGE_V3.equals( superType.toString() ) ) {
                methodName += LIST_SUFFIX;
            }

        }
        return methodName;
    }
```

(I also would recommend using constants).